### PR TITLE
Feature/add compressor recipes for nether brick and lump

### PR DESCRIPTION
**Pull request by gabriel1379**
Hi,

Another addition with nether, similar to the previous one, just this time for compressor.

Clears nether's default recipes for compressed nether brick and nether lump, using the compressor instead for these tasks.

+ did the same type of clearing up for better overview as with the grinder recipes in the other MR.

### CODE REVIEW

File: technic/machines/register/compressor_recipes.lua [hunk 1]
```diff
@@ -55,23 +38,92 @@ if minetest.get_modpath("everness") then
...
+		for _, craft_entry in ipairs(crafts) do
+			table.insert(crafts_to_clear, craft_entry)
+		end
 	end
 end
 
-for _, sand_name in ipairs(crafts_to_clear) do
+-- Clear recipes
+for _, craft_name in ipairs(crafts_to_clear) do
+	is_regular = string.sub(craft_name, 1, 7) ~= "nether:"
```
Comment (SmallJoker) [inline technic/machines/register/compressor_recipes.lua:60]: Please do not pollute the global namespace. Use `local is_regular`. Same for other occurrences such as `shaped_recipe`.

File: technic/machines/register/compressor_recipes.lua [hunk 2]
```diff
@@ -8,45 +8,28 @@ function technic.register_compressor_recipe(data)
...
-		{"everness:coral_white_sand 2",               "everness:coral_white_sandstone"},
-		{"everness:crystal_forest_deep_ocean_sand 2", "everness:crystal_forest_deep_ocean_sandstone_block"},
-		{"everness:crystal_sand 2",                   "everness:crystal_sandstone"},
-		{"everness:cursed_lands_deep_ocean_sand 2",   "everness:cursed_lands_deep_ocean_sandstone_block"},
-		{"everness:cursed_sand 2",                    "everness:cursed_sandstone_block"},
-		{"everness:mineral_sand 2",                   "everness:mineral_sandstone"},
-	}
+-- Function for cases where multiple mods exist with the same name,
+-- to check if we have the right one.
+local function is_right_mod(dependency)
```
Comment (SmallJoker) [inline technic/machines/register/compressor_recipes.lua:13]: Why exactly do we need this function? Uncraftable recipes do not pose any significant performance impact - at least not any more (fixed in recent Minetest versions). You could register the recipes as soon as `minetest.get_modpath` returns a non-nil value. That would also get rid of the `nether` dependency.

Comment (gabriel1379) [reply technic/machines/register/compressor_recipes.lua]: User "Desour" on my other pull request ( https://github.com/minetest-mods/technic/pull/638 ) recommended an additional check for making sure that the right nether mod is being used. But OK, removed it now.

However, the dependency is still needed as recipes from nether (the compressing ones) still need to be cleared too.

File: technic/machines/register/compressor_recipes.lua (hunk 1 above)
Comment (gabriel1379) [reply technic/machines/register/compressor_recipes.lua]: Right - I keep forgetting that this needs to be noted separately in Lua. Thanks for the reminder, made local.

File: technic/machines/register/compressor_recipes.lua (hunk 2 above)
Comment (gabriel1379) [reply technic/machines/register/compressor_recipes.lua]: Removed it from the other PR too just now.

**Review by SmallJoker — APPROVED** [review]
Looks good. Sandstone, desert stone and silver sandstone are still craftable.

Will merge in a few days unless there are objections.

**Pull request merged**